tower-timeout/src/lib.rs

```diff
@@ -83,50 +97,114 @@ where T: Future,
         match self.response.poll() {
             Ok(Async::Ready(v)) => return Ok(Async::Ready(v)),
             Ok(Async::NotReady) => {}
-            Err(e) => return Err(Error::Inner(e)),
+            Err(e) => return Err(Error(Kind::Inner(e))),
         }
 
         // Now check the sleep
         match self.sleep.poll() {
             Ok(Async::NotReady) => Ok(Async::NotReady),
-            Ok(Async::Ready(_)) => Err(Error::Timeout),
-            Err(_) => Err(Error::Timeout),
+            Ok(Async::Ready(_)) => Err(Error(Kind::Elapsed)),
+            Err(_) => Err(Error(Kind::Elapsed)),
```

Comment:
I realize this code was already "wrong", but I believe this should be `Kind::Timer(err)`.

Reply:
Nice catch, fixed.